### PR TITLE
Drop Python 3.5 support

### PR DESCRIPTION
**Pull request by noliveleger**
Python 3.5 has reached its end of life in September 2020. Almost 2 years ago.  Moreover `master` already contains breaking changes in Python 3.5

### CODE REVIEW

File: helpers/config.py [hunk 1]
```diff
@@ -1194,21 +1195,17 @@ def __questions_dev_mode(self):
...
                     not self.__dict['kpi_dev_build_id'] or
                     self.__dict['kpi_path'] != kpi_path
                 ):
-                    build_id = '{prefix}{timestamp}'.format(
-                        prefix=self.get_prefix('frontend'),
-                        timestamp=str(int(time.time()))
-                    )
-                    self.__dict['kpi_dev_build_id'] = build_id
+                    prefix = self.get_prefix('frontend')
+                    timestamp = str(int(time.time()))
```
Comment (joshuaberetta) [inline helpers/config.py:1207]: Is it necessary to have the `str(int(time.time()))` everywhere if it's being put inside a formatted string anyway?
```python
>>> f'{str(int(time.time()))}' == f'{int(time.time())}'
True
```

File: helpers/setup.py [hunk 2]
```diff
@@ -134,26 +134,20 @@ def update_hosts(cls, dict_):
...
-                        '\n{routes}'
-                        '\n{end_sentence}'
-                        ).format(
-                bof=tmp_host.strip(),
-                start_sentence=start_sentence,
-                routes=routes,
-                end_sentence=end_sentence
+            public_domain_name = dict_['public_domain_name']
+            routes = (
+                f"{dict_['local_interface_ip']}  "
```
Comment (joshuaberetta) [inline helpers/setup.py:139]: Is there a reason for the double space here? I see it was there before.

File: helpers/cli.py [hunk 3]
```diff
@@ -44,14 +44,15 @@ def colored_print(cls, message, color=NO_COLOR):
 
     @classmethod
     def colorize(cls, message, color=NO_COLOR):
-        return '{}{}{}'.format(color, message, cls.NO_COLOR)
+        return f'{color}{message}{cls.NO_COLOR}'
```
Comment (joshuaberetta) [inline helpers/cli.py:47]: If I run:
```bash
python2.7 run.py
```
I get this:
```
Traceback (most recent call last):
  File "run.py", line 6, in <module>
    from helpers.cli import CLI
  File "/tmp/kobo-install/helpers/cli.py", line 47
    return f'{color}{message}{cls.NO_COLOR}'
                                           ^
SyntaxError: invalid syntax
```
Looks like we need to keep `CLI` friendly with <3.5 to at least display the warning:
```
josh@work:/tmp/kobo-install$ git checkout origin/master -- helpers/cli.py
josh@work:/tmp/kobo-install$ python2.7 run.py
╔════════════════════════════════════════════════════════════════════╗
║                                                                    ║
║ Your Python version has reached the end of its life.Please upgrade ║
║ it as it is not maintained anymore.                                ║
║                                                                    ║
╚════════════════════════════════════════════════════════════════════╝
```

File: run.py [hunk 4]
```diff
@@ -4,11 +4,14 @@
...
+
+if (
+    sys.version_info[0] == 2
+    or (sys.version_info[0] == 3 and sys.version_info[1] <= 5)
+):
     message = (
-        'Python 2.7 has reached the end of its life on '
-        'January 1st, 2020. Please upgrade your Python as Python 2.7 is '
-        'not maintained anymore.'
+        'Your Python version has reached the end of its life.'
```
Comment (joshuaberetta) [inline run.py:13]: Missing a space at the end of the line:
```
'Your Python version has reached the end of its life. '
```

File: helpers/cli.py [hunk 5]
```diff
@@ -133,9 +134,10 @@ def get_message_with_default(cls, message, default):
             )
 
         if message:
-            message = '{}: '.format(message.strip()) if not default else message
+            message = message.strip()
+            message = f'{message}: ' if not default else message
```
[the remark this answers is not in the document]
Comment (noliveleger) [reply helpers/cli.py:138]: Let's keep it the way is was!

File: helpers/setup.py (hunk 2 above)
Comment (joshuaberetta) [reply helpers/setup.py:139]: Sounds good 👍

**Review by joshuaberetta — APPROVED** [review]

**Pull request merged**